src/main/java/io/split/android/client/events/executors/SplitEventExecutorFactory.java

```diff
@@ -0,0 +1,30 @@
+package io.split.android.client.events.executors;
+
+
+import io.split.android.client.events.SplitEvent;
+import io.split.android.client.events.SplitEventTask;
+
+/**
+ * Created by sarrubia on 4/6/18.
+ */
+
+public class SplitEventExecutorFactory {
+
+
+    public static SplitEventExecutorAbstract factory(SplitEvent event, SplitEventTask task, SplitEventExecutorResources resources){
+
+        SplitEventExecutorAbstract executor = null;
+
+        switch(event){
+            case SDK_READY:
```

Comment:
SDK_READY,SDK_READY_TIMED_OUT:
default:
  throw new IlegalArgumentException

Reply:
This exception shouldn't be thrown because the Factory is only for internal use and the arguments are not provided by the user

Reply:
in any case its good even if it is not used. When developing, if we forget about this case, it will immediately show up during development instead of silently no to something.
Also in java since you are doing the same for both cases you can do:
```
switch(event) {
   case SDK_READY,SDK_READY_TIMED_OUT:
      ...
   break
}
```

Reply:
same